### PR TITLE
feat(gripper): Add gripper support

### PR DESCRIPTION
**Pull request by ahiuchingau**
Gripper board bringup, adding some basic tasks

Monorepo pr to add node id is [here](https://github.com/Opentrons/opentrons/pull/9647#partial-pull-merging)

### CODE REVIEW

File: gripper/firmware/interfaces.cpp [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    motor_queue("Motor Queue");
+
+/**
+ * The motor struct.
+ */
+static motor_class::Motor motor{
+    spi_comms,
+    lms::LinearMotionSystemConfig<lms::BeltConfig>{
+        .mech_config =
+            lms::BeltConfig{.belt_pitch = 2, .pulley_tooth_count = 20},
```
Comment (sfoster1) [inline gripper/firmware/interfaces.cpp:119]: it's a lead screw isn't it?

Comment (ahiuchingau) [reply gripper/firmware/interfaces.cpp:119]: yes will change it once i get the vals from hardware

File: gripper/core/tasks.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+#include "motor-control/core/tasks/motion_controller_task_starter.hpp"
+#include "motor-control/core/tasks/motor_driver_task_starter.hpp"
+#include "motor-control/core/tasks/move_group_task_starter.hpp"
+#include "motor-control/core/tasks/move_status_reporter_task_starter.hpp"
+
+static auto tasks = gripper_tasks::AllTask{};
+static auto queues = gripper_tasks::QueueClient{can_ids::NodeId::gripper};
+
+static auto mc_task_builder =
+    motion_controller_task_starter::TaskStarter<lms::BeltConfig, 512,
```
[the remark this answers is not in the document]
Comment (ahiuchingau) [reply gripper/core/tasks.cpp:13]: oui

File: gripper/firmware/main.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#include <array>
+#include <cstdio>
```
Comment (amitlissack) [inline gripper/firmware/main.cpp:2]: cstdio? and cstring?

File: gripper/firmware/motor_hardware.c [hunk 4]
```diff
@@ -0,0 +1,163 @@
+#include "motor_hardware.h"
+
+#include "gantry/core/axis_type.h"
```
Comment (amitlissack) [inline gripper/firmware/motor_hardware.c:3]: Is this needed?

File: gripper/firmware/motor_hardware.h [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+extern "C" {
+#endif  // __cplusplus
+
+extern SPI_HandleTypeDef hspi2;
+extern TIM_HandleTypeDef htim7;
+
+typedef void (*motor_interrupt_callback)();
+
+HAL_StatusTypeDef initialize_spi();
+void gantry_driver_CLK_init(enum GantryAxisType);
```
Comment (amitlissack) [inline gripper/firmware/motor_hardware.h:17]: gantry?

File: gripper/firmware/motor_hardware.h [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "common/firmware/errors.h"
+#include "gantry/core/axis_type.h"
```
Comment (amitlissack) [inline gripper/firmware/motor_hardware.h:4]: gantry?

File: gripper/simulator/interfaces.cpp [hunk 7]
```diff
@@ -0,0 +1,73 @@
+#include "gantry/core/interfaces.hpp"
+
+#include "can/simlib/sim_canbus.hpp"
+#include "can/simlib/transport.hpp"
+#include "common/simulation/spi.hpp"
+#include "gantry/core/tasks.hpp"
```
Comment (amitlissack) [inline gripper/simulator/interfaces.cpp:6]: gantry?

File: gripper/simulator/interfaces.cpp [hunk 8]
```diff
@@ -0,0 +1,73 @@
+#include "gantry/core/interfaces.hpp"
```
Comment (amitlissack) [inline gripper/simulator/interfaces.cpp:1]: Gantry?

File: gripper/simulator/main.cpp [hunk 9]
```diff
@@ -0,0 +1,14 @@
+#include "FreeRTOS.h"
+#include "gantry/core/interfaces.hpp"
```
Comment (amitlissack) [inline gripper/simulator/main.cpp:2]: Gantry?

File: gripper/simulator/interfaces.cpp [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+                                      .min_acceleration = 1,
+                                      .max_acceleration = 2},
+    utils::driver_config(),
+    motor_queue};
+
+/**
+ * Handler of motor interrupts.
+ */
+static motor_handler::MotorInterruptHandler motor_interrupt(
+    motor_queue, gantry_tasks::get_queues(), motor_interface);
```
Comment (amitlissack) [inline gripper/simulator/interfaces.cpp:54]: Gantry again :(

File: gripper/tests/CMakeLists.txt [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+include(CTest)
+include(Catch)
+include(AddBuildAndTestTarget)
+
+add_executable(
+        gripper
+        test_main.cpp
+)
+
+target_include_directories(gantry PUBLIC ${CMAKE_SOURCE_DIR}/include)
```
Comment (amitlissack) [inline gripper/tests/CMakeLists.txt:13]: gantry

**Review by amitlissack — COMMENTED** [review]
A couple more gantry references?

File: gripper/simulator/interfaces.cpp (hunk 10 above)
Comment (ahiuchingau) [reply gripper/simulator/interfaces.cpp:54]: Sometimes I wonder if I'm actually blind... Thanks for finding these @amitlissack

**Review by amitlissack — APPROVED** [review]

**Pull request merged**